pyanaconda/payload/__init__.py

```diff
@@ -1272,8 +1275,11 @@ def _setup_harddrive_device(self, storage, method, isodev, device):
             if not isodevice:
                 raise PayloadSetupError("device for HDISO install %s does not exist" % devspec)
 
-            self._setupMedia(isodevice)
-            url = "file://" + INSTALL_TREE
+            if self._setupMedia(isodevice):
+                url = "file://" + INSTALL_TREE + method.dir
+            else:
+                url = "file://" + INSTALL_TREE
```

Comment:
Couldn't you return the url instead of the boolean value? This solution looks like a hack to me.